upgrade/upgrade-scripts/common.sh

```diff
@@ -193,6 +193,17 @@ function copy_dataset_property() {
 		die "failed to set property '$PROP_NAME=$PROP_VALUE' for '$DST_DATASET'"
 }
 
+function copy_dataset_property_to_snapshots_recursive() {
```

Comment:
I don't feel strongly, but I would prefer to inline this function into the callers. By doing so, we'd duplicate a small amount of code (a loop, and one call to `zfs list`), but I think that'll make it more clear to any readers/maintainers of these scripts exactly what is happening.

What do you think?

I'll approve as-is in case you'd rather stick with this function.